### PR TITLE
contributing.rst: update and expand contributing guide

### PR DESCRIPTION
**Pull request by grondo**
Update contributing guide to remove some generalities with specifics, add examples, etc.
Includes:

 * Reference GitHub "fork and pull" model in description of PRs
 * mention pr-validator script
 * Mention that contributions should include relevant tests in Developer Guidelines
 * Give examples of "good"  `subsys:` prefix names, include reason for subsystem prefix in commit messages
 * Note directly that commit messages should be wrapped at about 72 characters

This was done quickly, so I apologize in advance for any dumb mistakes.

### CODE REVIEW

File: contributing.rst [hunk 1]
```diff
@@ -14,7 +14,18 @@ This guide details how to contribute to `Flux Framework projects <https://github
 Pull Requests
 -------------
 
-Use a pull request (PR) toward a repository's master branch to propose your contribution, including the specific issue number your PR resolves. If you are planning significant code changes, or have any questions, please open an issue and reference it in your PR. To contribute to a Flux Framework project:
+Pull requests are the primary way code changes are incorporated into
+Flux Framework projects. All projects use the GitHub
+`fork and pull <https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-collaborative-development-models>`
+model, in which contributors develop on a branch of their personal
+fork of the project and create pull requests in order to requests
```
Comment (SteVwonder) [inline contributing.rst]:
```suggestion
fork of the project and create pull requests in order to request
```

Comment (grondo) [reply contributing.rst]: Haha, sorry, will fix! :woman_facepalming:

**Comment by grondo** [comment]
Ok, I made the change suggested by @SteVwonder. But checking the read-the-docs build, I didn't get the syntax right for adding a link in reStructuredText. Will try again.

**Comment by grondo** [comment]
I had butchered the reStructuredText -- fixed up and re-pushed. Sorry for the noise @SteVwonder!

**Pull request merged**